src/pipx/commands/upgrade.py

```diff
@@ -34,21 +32,44 @@ def upgrade(
 
     venv = Venv(venv_dir, verbose=verbose)
 
-    old_version = venv.get_venv_metadata_for_package(package).package_version
+    package_metadata = venv.package_metadata[package]
+    if package_metadata.package_or_url is not None:
+        package_or_url = package_metadata.package_or_url
+        old_version = package_metadata.package_version
+        include_apps = package_metadata.include_apps
+        include_dependencies = package_metadata.include_dependencies
+    else:
+        # fallback if no metadata
+        package_or_url = package
+        old_version = ""
+        include_apps = True
+        include_dependencies = False
+
+    if package == "pipx":
+        package_or_url = PIPX_PACKAGE_NAME
```

Comment:
The `PIPX_PACKAGE_NAME` constant was introduced before we got the package name `pipx` on PyPI, when it was `pipxapp`. This code isn't necessary, we can just use `pipx` and not `PIPX_PACKAGE_NAME`.

Reply:
Ok thanks for clearing this up!  I was a little mystified about what was going on here.
Done.